### ISSUE TITLE
Sound bug issue: the game "Blackthorne" hangs and refuses to load

### ISSUE THREAD

**Issue by Banjo-Oz** [joncampbell123/dosbox-x#2402]
This happens in both DOSBox-X and Staging. It doesn't happen in ECE.

When I load the game (or demo) of the Inteprlay game "Blackthorne", it hangs on the loading message before starting the game... forever. This does NOT happen if I use the included setup program to specify "no sound" for sound effects. Music settings don't cause a problem (I can set Sound Blaster, General Midi, none, etc.) but digital sound effects cause this issue. I have double checked that my sound blaster IRQ and DMA settings match in both the setup for the game and my DOSBox-X config file. I am fairly certain it's a bug and not user error since this does not happen in DOSBox-ECE or "vanilla" DOSBox.

**Comment by Wengier** [comment]
Thanks for reporting this! I can confirm the issue with the said game, and indeed the issue does not occur when I select "No sound" for the Setup program. I suspect the issue is caused by the Sound Blaster emulation code for the sound effects.

**Comment by joncampbell123** [comment]
I am able to use Sound Blaster Pro digitized sound just fine. I am using this version: https://www.myabandonware.com/game/blackthorne-2p8#download

[cpu]
core=normal
cputype=386
cycles=10000

[sblaster]
#sbtype=sb16
sbtype=sbpro2
#sbtype=sb16vibra
sbbase=220
irq=5
dma=1
hdma=5

[autoexec]
mount c: .
mount d: /usr/src/doslib
c:
setup
bthorne

https://user-images.githubusercontent.com/6245486/113098695-ad6e7880-91ad-11eb-917d-1efa8e8117db.mp4

**Comment by joncampbell123** [comment]
It also works with sbtype=sb16. Usually if a game doesn't seem to work with Sound Blaster, it might have code that requires Sound Blaster 2.0 or Sound Blaster Pro.

**Comment by Wengier** [comment]
@Banjo-Oz Yes, it turns out that the cycles is the issue for this. For example, when I select "386DX 25MHz (~4595 cycles)" or above from "Emulate CPU speed" menu group ("under "CPU") it will run, but not options below it.

I also tried vanilla DOSBox with "cycles=3000" and it does not appear to work either. Setting a higher value (e.g. "cycles=5000") then the game will load.

**Comment by Banjo-Oz** [comment]
@Wengier That makes sense re: vanilla; I only ever tried dropping it to 6000.

So am I understanding right that "auto" is assuming 3000 wrongly in DOSBox-X whereas vanilla "auto" sets it correctly (higher) and thus it works?

**Comment by Wengier** [comment]
@Banjo-Oz I have tried DOSBox 0.74.3 and DOSBox ECE r4441, and it appears that setting "core=normal" and "cycles=auto" also assumes 3000 cycles for them and the game does not run. But setting a higher cycle or "cycles=max" will work, which appears to be the same behavior as in DOSBox-X. Can you check it again?

**Comment by Banjo-Oz** [comment]
@Wengier I can confirm the assumed 3000 cycles (cycles=auto, core=auto) and thus **not** running with vanilla 0.74.3 either. My custom config kept overriding to 6000 and I didn't notice! Sorry! :(

Also just tried DOSBox-X 0.83.12 and issue remains with "auto" cycles.

**Comment by NebularNerd** [comment]
Just tested with a recent build `2022.09.01 autobuild` and you still need to manually set the cycles, otherwise game plays just fine.
